components/feature/prompts/src/main/java/mozilla/components/feature/prompts/dialog/LoginDialogFragment.kt

```diff
@@ -224,42 +233,42 @@ internal class LoginDialogFragment : PromptDialogFragment() {
         )
         // The below code is not thread safe, so we ensure that we block here until we are sure that
         // the previous call has been canceled
-        runBlocking { stateUpdate?.cancelAndJoin() }
+        runBlocking { validateStateUpdate?.cancelAndJoin() }
         var validateDeferred: Deferred<Result>? = null
-        stateUpdate = scope.launch(IO) {
-            validateDeferred = feature?.loginValidationDelegate?.validateCanPersist(login)
+        validateStateUpdate = scope.launch(IO) {
+            // We only want to fetch this list once. Since we are ignoring username the results will
+            // not change when user changes username or password fields
+            val listToCheck = if (potentialDupesList == null) {
+                this@LoginDialogFragment.potentialDupesList = CopyOnWriteArrayList(
+                    loginValidationDelegate.getPotentialDupesIgnoringUsernameAsync(
+                        login
+                    ).await()
+                )
+                potentialDupesList?.toList()
+            } else {
+                potentialDupesList?.toList()
+            }
+            validateDeferred =
```

Comment:
nit: move the `.toList()` on the invocation line, instead, makes the intent of passing in a copy of the list clearer. it's also really worth your time to leave a comment here explaining why we're passing in a copy, and not the original.

this will also let you simplify the above statement, removing the `else` clause.